src/main/user-impl/java/com/mysql/cj/jdbc/ha/ca/plugins/ConnectionPluginManager.java

```diff
@@ -106,8 +106,8 @@ public void init(ClusterAwareConnectionProxy proxy, PropertySet propertySet) {
 
   //TODO: Should method target to be passed along with method name? Like:
   // Object execute(Object methodTarget, String methodName, Callable executeSqlFunc)
-  public Object execute(String methodName, Callable executeSqlFunc) throws Exception {
-    return this.headPlugin.execute(methodName, executeSqlFunc);
+  public Object execute(Class methodInvokeOn, String methodName, Callable executeSqlFunc) throws Exception {
```

Comment:
```suggestion
  public Object execute(Class<?> methodInvokeOn, String methodName, Callable executeSqlFunc) throws Exception {
```

https://stackoverflow.com/questions/4337313/why-is-class-preferred-to-class

Reply:
Can we add `<?> to the Callable as well?